components/server/ee/src/billing/billing-mode.ts

```diff
@@ -0,0 +1,174 @@
+import { TeamSubscription2 } from "@gitpod/gitpod-protocol/lib/team-subscription-protocol";
+
+export const BillingModes = Symbol("BillingModes");
+export interface BillingModes {
+    getBillingMode(attributionId: AttributionId, now: Date): Promise<BillingMode>;
+    getBillingModeForUser(user: User, now: Date): Promise<BillingMode>;
+    getBillingModeForTeam(team: Team, now: Date): Promise<BillingMode>;
+}
+
+/**
+ *
+ * Some rules for how we decide about BillingMode someone is in:
+ *  - Teams: Do they have either:
+ *    - Chargebee subscription              => cb
+ *    - UBB (& maybe Stripe subscription):  => ubb
+ *  - Users: Do they have either:
+ *    - personal Chargebee subscription     => cb
+ *    - personal Stripe Subscription        => ubb
+ *    - at least one Stripe Team seat       => ubb
+ */
+@injectable()
+export class BillingModesImpl implements BillingModes {
+    @inject(Config) protected readonly config: Config;
+    @inject(ConfigCatClientFactory) protected readonly configCatClientFactory: ConfigCatClientFactory;
+    @inject(SubscriptionService) protected readonly subscriptionSvc: SubscriptionService;
+    @inject(StripeService) protected readonly stripeSvc: StripeService;
+    @inject(TeamSubscription2DB) protected readonly teamSubscription2Db: TeamSubscription2DB;
+    @inject(TeamDB) protected readonly teamDB: TeamDB;
+    @inject(UserDB) protected readonly userDB: UserDB;
+
+    async getBillingMode(attributionId: AttributionId, now: Date): Promise<BillingMode> {
+        switch (attributionId.kind) {
+            case "team":
+                const team = await this.teamDB.findTeamById(attributionId.teamId);
+                if (!team) {
+                    throw new Error(`Cannot find team with id '${attributionId.teamId}'!`);
+                }
+                return this.getBillingModeForTeam(team, now);
+            case "user":
+                const user = await this.userDB.findUserById(attributionId.userId);
+                if (!user) {
+                    throw new Error(`Cannot find user with id '${attributionId.userId}'!`);
+                }
+                return this.getBillingModeForUser(user, now);
+        }
+    }
+
+    async getBillingModeForUser(user: User, now: Date): Promise<BillingMode> {
+        if (!this.config.enablePayment) {
+            // Payment is not enabled. E.g. Self-Hosted.
+            return { mode: "none" };
+        }
+
+        // Is Usage Based Billing enabled for this user or not?
+        const teams = await this.teamDB.findTeamsByUser(user.id);
+        const isUsageBasedBillingEnabled = await this.configCatClientFactory().getValueAsync(
+            "isUsageBasedBillingEnabled",
+            false,
+            {
+                user,
+                teams,
+            },
+        );
+
+        // 1. UBB enabled?
+        if (!isUsageBasedBillingEnabled) {
+            // No Chargebee subscription, and UBB is not enabled: chargebee (free)
```

Comment:
Nit: This comment is inaccurate (we don't know that here are no chargebee subscriptions because we didn't check yet)